docs/commercial/single-sign-on.rst

```diff
@@ -0,0 +1,90 @@
+   If you would like to apply for the Beta, please `contact us <mailto:support@readthedocs.com>`_.
+
+.. contents::
+   :local:
+   :depth: 2
+
+
+SSO with GitHub, Bitbucket or GitLab
+------------------------------------
+
+Using an Identity Provider that supports authentication and authorization allows you to manage
+"who have access to what projects on Read the Docs" directly from the provider itself.
+In case you want a user to have access to your documentation project under Read the Docs,
+that user just needs to be granted permissions in the GitHub, Bitbucket or GitLab repository associated with it.
+
+Note the users created under Read the Docs must have their GitHub, Bitbucket or GitLab
+account connected in order to make SSO to work.
+
+.. note::
+
+   You can read more about `granting permissions on GitHub`_.
+
+   .. _granting permissions on GitHub: https://docs.github.com/en/github/setting-up-and-managing-organizations-and-teams/repository-permission-levels-for-an-organization
+
+
+Grant access to read the documentation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By granting **read** (or more) permissions to a user under GitHub, Bitbucket or GitLab
+you are giving access to read the documentation of the associated project on Read the Docs to that user.
+
+
+Grant access to administrate a project
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By granting **write** permission to a user under GitHub, Bitbucket or GitLab
+you are giving access to read the documentation *and* to be an administrator
+of the associated project on Read the Docs to that user.
+
+
+Grant access to import a project
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When SSO with GitHub, Bitbucket or GitLab is enabled in the organization only owners can import projects on the organization.
+Adding users as owners of your organization will give them permissions to import projects.
+
+Note that to be able to import a project, that user must have **admin** permissions in the GitHub, Bitbucket or GitLab repository associated.
+
+
+SSO with a ``@company.com`` email address
+-----------------------------------------
+
+Using a ``@company.com`` email address allows you to
+"grant **read** access to all the projects under your organization to users with a ``@company.com`` verified email address".
+
+
+Grant access to administrate a project
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can add a user under an "Admin Team" to grant admin permissions to all the projects under that Team.
+This can be done under :guilabel:`Teams` > :guilabel:`Admins` > :guilabel:`Invite Member`.
+
+
+Grant access to import a project
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Making the user member of any "Admin Team" under your organization (as mentioned in the previous section),
+they will be granted access to import a project.
+
+Note that to be able to import a project, that user must have at least **read** permissions in the GitHub, Bitbucket or GitLab repository associated,
```

Comment:
This seems wrong, no? We can't set webhooks on these repos, and I think it isn't how it works currently right?

Reply:
Right. This is a mistake.

Reply:
Wait... This is confusing.

Currently, _without SSO_, if the project is public and you have **read** permissions on it, you can imported it, see https://github.com/readthedocs/readthedocs.org/blob/eb34b08ee05acc9f3c4df0a4dd1fed2422a998df/readthedocs/projects/static-src/projects/js/import.js#L123-L128 . In that case, you will be able to import the project, but the webhook won't be connected, ssh key not setup and we will show these notifications:

![Screenshot_2020-07-07_19-32-25](https://user-images.githubusercontent.com/244656/86819924-9bc9cb80-c088-11ea-8df5-f9a460410eb2.png)

I changed that behavior to only make `is_locked = !self.admin()` only when SSO is enabled because it doesn't make sense to import a project that you are not able to modify later (GitHub SSO won't give you access to Admin tab if you don't have **admin** permissions in the GH repo).

Summarizing,

* current behavior with RTD Auth, allows you to import public projects with only **read** permissions
* GitHub SSO, allows you to import projects _only_ if you **admin** on the GH repository
* Verified Email, same as RTD Auth

I think, for corporate, it makes sense to _always_ force to be admin on the GH repository to be able to import a project; because the reasons that you mentioned. After imported, you will end up with a half configured project. This would change the current behavior, tho.

Reply:
I changed this behavior to allow import projects only with `admin` permissions in the repository for Email SSO and VCS SSO.

I think, we should eventually do the same for RTD Auth and make all behave the same under corporate: "admin permissions on RemoteRepository are required to import a project".